### PR TITLE
Fix control operator for for-loop

### PR DESCRIPTION
**Pull request by DarkmatterVale**
As requested by @mrodriguez-parallax in #465

### CODE REVIEW

**Comment by PropGit** [comment]
... reviewing...

File: src/main/webapp/cdn/blockly/generators/propc/control.js [hunk 1]
```diff
@@ -106,10 +106,10 @@ Blockly.propc.control_repeat_for_loop = function () {
     var loop_counter = Blockly.propc.variableDB_.getName(this.getFieldValue('VAR'), Blockly.Variables.NAME_TYPE);
 
     if (Number(start) < Number(end)) {
-        var code = 'for (' + loop_counter + ' = ' + start + '; ' + loop_counter + ' < ' + end + '; ' + loop_counter + ' += ' + step + ') {\n' + repeat_code + '\n}';
+        var code = 'for (' + loop_counter + ' = ' + start + '; ' + loop_counter + ' <= ' + end + '; ' + loop_counter + ' += ' + step + ') {\n' + repeat_code + '\n}';
         return code;
     } else {
-      var code = 'for (' + loop_counter + ' = ' + start + '; ' + loop_counter + ' > ' + end + '; ' + loop_counter + ' += ' + step + ') {\n' + repeat_code + '\n}';
+      var code = 'for (' + loop_counter + ' = ' + start + '; ' + loop_counter + ' >= ' + end + '; ' + loop_counter + ' += ' + step + ') {\n' + repeat_code + '\n}';
```
Comment (PropGit) [inline src/main/webapp/cdn/blockly/generators/propc/control.js]: Ln 112 - If start => end, then the condition __loop_counter >= end__ will _always_ be true when the modifier __loop_counter += step__ has a positive _step_ value.

**Comment by PropGit** [comment]
I may have spotted a logical error.  See [line comment](https://github.com/parallaxinc/BlocklyProp/pull/619/commits/279ac60c31bc414a878e901312d31551c624eaa7).

**Comment by DarkmatterVale** [comment]
@PropGit Good catch; I made a change to the increment so the logic error should be gone.

**Comment by PropGit** [comment]
@DarkmatterVale - Sure!  Question: Is _step_ itself guaranteed to be a positive value?

**Comment by DarkmatterVale** [comment]
Hmm...No, not at the moment it is not. Do you think I should take the absolute value?

**Comment by PropGit** [comment]
Yes, I think so, on both versions of the statement, to ensure the index occurs in the direction implied by the relative start/end values.  Looks like this one has been merged in already.

**Comment by DarkmatterVale** [comment]
Gotcha; I will send another PR with the change.

**Comment by DarkmatterVale** [comment]
How do I do absolute value in propeller C? Is it just abs()?

**Comment by PropGit** [comment]
@DarkmatterVale - Yes abs() should do it.  Standard C Library stuff, not implemented separately in Simple Libraries to my knowledge.

**Comment by DarkmatterVale** [comment]
Ah ok, I didn't know that all the normal functions were already included...

**Pull request merged**